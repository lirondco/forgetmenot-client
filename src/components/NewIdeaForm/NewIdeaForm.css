.insertIdea {
    padding: 5%;
}

.insertIdea h3 {
    text-align: center;
    margin: 5%;
    font-family: 'Patrick Hand SC', cursive;
    font-size: 21pt;
}

.insertIdea form {
    display: flex;
    flex-direction: column;
}

.insertIdea form select {
    margin-bottom: 5%;
    padding: 10px;
    width: fit-content;
}