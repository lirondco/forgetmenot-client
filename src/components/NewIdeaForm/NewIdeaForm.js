import React, { Component } from 'react'
import IdeasContext from '../../contexts/IdeasContext'
import IdeasApiService from '../../services/ideas-api-service'
import { Input, Button, Textarea, Required } from '../Utils/Utils'
import { Link } from 'react-router-dom'
import './NewIdeaForm.css'

export default class NewIdeaForm extends Component {  
  static defaultProps = {
      onPostIdeaSuccess: () => {}
  }
  static contextType = IdeasContext
  listSelection = []
  missingList = false

  componentDidMount() {
    IdeasApiService.getLists()
      .then(this.context.setLists)
      .then(this.context.clearError)
      .catch(this.context.setError)
  }

  handleSubmit = ev => {
    ev.preventDefault()
    const { list } = ev.target
    const { name } = ev.target
    const { content } = ev.target

    IdeasApiService.postIdea(list.value, name.value, content.value)
      .then(() => {
        list.value = ''
        name.value = ''
        content.value = ''
        this.props.onPostIdeaSuccess()
      })
      .catch(this.context.setError)
      .catch(this.missingList = true)
  }

  renderListError() {
    console.log(this.missingList)
    if(this.missingList === true) {
      return (
        <p className="red">Please select a list! If you don't have a list, <Link to='/newlist'>click here</Link> to make one.</p>
      )
    }
  }

  render() {
    const { lists=[] } = this.context
    return (
    <div className='insertIdea'>
    <h3>You're entering a new idea</h3>
      <form
        className='ideaForm'
        onSubmit={this.handleSubmit}
      >
        <div className='ideaName'>
            <label htmlFor='name'>New idea's name:</label>
            <Required />
            <Input
                required
                type='text'
                maxLength='30'
                name='name'
                id='name'
                />
        </div>
        <div className='content'>
          <label htmlFor='content'>
            Description
          </label>
          <Required />
          <Textarea
            required
            name='content'
            id='content'
            cols='30'
            rows='3'
            maxLength='240'
            placeholder='Type a short description...'>
          </Textarea>
          <br />
          <label htmlFor='list'>Add to which list?</label>
        <Required />
        <select id='list' name='list' required>
            <option value={null}>...</option>
            {lists.map(list => {
                return (
                <option key={list.id} value={list.id}>
                    {list.name}
                </option>
                )
            })}
        </select>
        </div>
        <Button type='submit'>
          Add Idea
        </Button>
        {this.renderListError()}
      </form>
    </div>
    )
  }
}
