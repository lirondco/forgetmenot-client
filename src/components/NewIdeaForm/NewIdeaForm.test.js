import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewIdeaForm from './NewIdeaForm'

describe(`NewIdeaForm component`, () => {
  const stubLists = [
    {
      "id": 1,
      "name": "Test A"
    },
    {
      "id": 2,
      "name": "Test B"
    },
    {
      "id": 3,
      "name": "Test C"
    }
  ]


  it('renders the complete form', () => {
    const context = { clearError: () => {} }
    const wrapper = shallow(<NewIdeaForm />, context)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the select options from lists', () => {
    const context = stubLists
    const select = shallow(<NewIdeaForm />, context)
      .find('#list')
    expect(toJson(select)).toMatchSnapshot()
  })
})
