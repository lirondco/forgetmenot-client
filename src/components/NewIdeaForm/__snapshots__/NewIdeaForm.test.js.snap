// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIdeaForm component renders the complete form 1`] = `
<div
  className="insertIdea"
>
  <h3>
    You're entering a new idea
  </h3>
  <form
    className="ideaForm"
    onSubmit={[Function]}
  >
    <div
      className="ideaName"
    >
      <label
        htmlFor="name"
      >
        New idea's name:
      </label>
      <Required />
      <Input
        id="name"
        maxLength="30"
        name="name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="content"
    >
      <Required />
      <Textarea
        aria-label="Type a short description..."
        cols="30"
        id="content"
        maxLength="240"
        name="content"
        placeholder="Type a short description..."
        required={true}
        rows="3"
      />
    </div>
    <label
      htmlFor="list"
    >
      Add to which list?
    </label>
    <Required />
    <select
      id="list"
      name="list"
      required={true}
    >
      <option
        value={null}
      >
        ...
      </option>
    </select>
    <Button
      type="submit"
    >
      Add Idea
    </Button>
  </form>
</div>
`;

exports[`NewIdeaForm component renders the select options from lists 1`] = `
<select
  id="list"
  name="list"
  required={true}
>
  <option
    value={null}
  >
    ...
  </option>
</select>
`;
